app/api/create-agent/route.ts

```diff
@@ -0,0 +1,47 @@
+import { NextResponse } from "next/server";
+import { z, ZodError } from "zod";
+import { cookies } from "next/headers";
+import { spawn } from "child_process";
+
+const createAgentSchema = z.object({
+  name: z.string().min(1),
+});
+
+export async function POST(request: Request) {
+  try {
+    const cookieStore = await cookies();
+    const token = cookieStore.get("admin-token");
+
+    if (token?.value !== process.env.ADMIN_TOKEN) {
+      return NextResponse.json({ error: "Unauthorized" }, { status: 401 });
+    }
+
+    const body = await request.json();
+    const { name } = createAgentSchema.parse(body);
+    
+    const createProcess = spawn('npm', ['run', 'create-agent', name], {
+      detached: true,
+      stdio: 'ignore'
+    });
```

Comment:
Instead of running an npm command like this, manually workflow dispatch a GitHub action for create agent using the GitHub api